src/plugins/vis_type_timeseries/server/lib/timeseries_visualization_client_wrapper.ts

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import {
+  SavedObjectsClientContract,
+  SavedObjectsClientWrapperFactory,
+  SavedObjectsClientWrapperOptions,
+  SavedObjectsCreateOptions,
+} from '../../../../core/server';
+import { getDataSourceEnabled } from './services';
+
+export const TIMESERIES_VISUALIZATION_CLIENT_WRAPPER_ID = 'timeseries-visualization-client-wrapper';
+
+export const timeSeriesVisualizationClientWrapper: SavedObjectsClientWrapperFactory = (
+  wrapperOptions: SavedObjectsClientWrapperOptions
+) => {
+  const createForTimeSeries = async <T = unknown>(
+    type: string,
+    attributes: T,
+    options?: SavedObjectsCreateOptions
+  ) => {
+    if (type !== 'visualization' || !getDataSourceEnabled().enabled) {
+      return await wrapperOptions.client.create(type, attributes, options);
+    }
+
+    // @ts-expect-error
+    const visState = JSON.parse(attributes.visState);
+
+    if (visState.type !== 'metrics' || !visState.params) {
+      return await wrapperOptions.client.create(type, attributes, options);
+    }
+
```

Comment:
Hi Huy I am really new to fix TSVB, could you help me to understand the logic here?  
I see the first condition `(type !== 'visualization' || !getDataSourceEnabled().enabled)` and ` (visState.type !== 'metrics' || !visState.params) ` both returns  `return await wrapperOptions.client.create(type, attributes, options); `  Can we combine it here?

Reply:
So only when type is metrics and visualization, then we consider adding datasource?

Reply:
So this client wrapper can receive any type of saved object and only Visualization objects will have the `visState` property. Thus, the first condition checks if the object is a visualization object (or if MDS is disabled) before proceeding. The second condition then checks that the visualization object has `params` and is a `metrics` (TSVB) type.